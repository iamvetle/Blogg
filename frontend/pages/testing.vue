<template>
  <div class="flex">

    <form @submit.prevent method="post" class="flex flex-col" enctype="multipart/form-data">

      <label for="name">Name:</label>
      <input type="text" name="name" id="name">

      <label for="email">Email</label>
      <input type="email" name="email" id="email">

      <label for="message">Message:</label>
      <textarea name="message" id="" cols="30" rows="10" placeholder="Write a message here"></textarea>

      <label for="file">File:</label>
      <input type="file" name="file" id="file">

      <button type="submit">Submit</button>
    </form>

  </div>
</template>

<script setup lang="ts">
onMounted(() => {

  const form = document.querySelector("form")

  form?.addEventListener("submit", (e: any) => {


    const formData = new FormData(form)

    for (let item of formData) {
      console.log(item[0], item[1])
    }
    
    // ta axios of prÃ¸v fetch her
  })





})

// watchEffect(() => {

// })

</script>

<style scoped></style>