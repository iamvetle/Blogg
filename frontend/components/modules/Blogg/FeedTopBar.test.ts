import FeedTopBar from './FeedTopBar.vue';
import { shallowMount } from '@vue/test-utils';
import { describe, expect, test, beforeEach, afterEach } from 'vitest';
import { createTestingPinia } from '@pinia/testing';


let wrapper: VueWrapper;
let pinia: any = createTestingPinia();

// let generalStore; 
let postStore;
let loggedInUserStore;
let paginationStore:any;
let authStore:any

let mockSetToOnlyShowFollowingPosts = vi.fn()
let mockShowAllFeedPosts = vi.fn()

const factory = () => {
	return shallowMount(FeedTopBar, {
		global: {
			plugins: [pinia],
			components: {
			},
			mocks: {
				handleShowFollowingPosts: mockSetToOnlyShowFollowingPosts,
				handleShowAllFeedPosts: mockShowAllFeedPosts,
			},
			stubs: {
                "FeedTopChoice":true
			},
		},
		slots: {}
	})
};

describe('Testing component', () => {

	beforeEach(() => {
		// generalStore = useGeneralStore(pinia); 
		postStore = usePostStore(pinia);
		loggedInUserStore = useLoggedInUserStore(pinia);
		paginationStore = usePaginationStore(pinia); 
		authStore = useAuthStore(pinia)

		paginationStore.activeFetchURL = "http://localhost:8888/api/feed/following/"


	});

	afterEach(() => {
		if (wrapper) {
			wrapper.unmount();
		}
		vi.clearAllMocks()
	});
	test('Should exist', async () => {
		wrapper = factory()
		authStore.isAuthenticated = true

		expect(wrapper.exists()).toBe(true)
	})

	/** The web client should not get any options when not authenticated */
	test('Should match snapshot when not authenticated', async () => {
		wrapper = factory()
		authStore.isAuthenticated = false;

		await wrapper.vm.$nextTick()
	
		expect(wrapper.html()).toMatchSnapshot()
	})
	test('Should match snapshot when authenticated', async () => {
		wrapper = factory()
		authStore.isAuthenticated = true
		await wrapper.vm.$nextTick()
		
		expect(wrapper.html()).toMatchSnapshot()
	})
	test('Should render the COMPONENT "FeedTopChoice"', async () => {
		wrapper = factory()
		authStore.isAuthenticated = true
		await wrapper.vm.$nextTick()

		expect(wrapper.findComponent("FeedTopChoice").exists()).toBe(true)
	})
	test('Should render the following option button when authenticated', async () => {
		wrapper = factory()
		authStore.isAuthenticated = false
		await wrapper.vm.$nextTick()

		expect(wrapper.findComponent("FeedTopChoice").exists()).toBe(false)
	})

	test('Should RENDER the FILTER COMPONENT regardless of if the user is authenticated or not', async () => {
		wrapper = factory()
		authStore.isAuthenticated = true
		await wrapper.vm.$nextTick()
		
		expect(wrapper.findComponent("FeedDropdownFilter").exists()).toBe(true)

		wrapper.unmount()

		wrapper = factory()
		authStore.isAuthenticated = false
		await wrapper.vm.$nextTick()
		
		expect(wrapper.findComponent("FeedDropdownFilter").exists()).toBe(true)
	})
    
});