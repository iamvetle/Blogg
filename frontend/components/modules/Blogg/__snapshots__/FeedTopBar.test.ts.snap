// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing component > Should match snapshot when authenticated 1`] = `
"<div class=\\"flex-row\\">
  <div class=\\"grid gap-8 grid-cols-12 flex items-center\\">
    <div class=\\"col-start-1 col-end-9\\">
      <!-- Only show the two options when web client is authenticated-->
      <div class=\\"text-center space-x-6\\">
        <feedtopchoice></feedtopchoice>
      </div>
    </div>
    <div class=\\"col-span-4\\" data-test=\\"feed_filter_component\\">
      <feeddropdownfilter data-test=\\"feed_dropdown_filter\\"></feeddropdownfilter>
    </div>
  </div><!-- IF the following buttons is pressed-->
  <!--v-if-->
</div>"
`;

exports[`Testing component > Should match snapshot when not authenticated 1`] = `
"<div class=\\"flex-row\\">
  <div class=\\"grid gap-8 grid-cols-12 flex items-center\\">
    <div class=\\"col-start-1 col-end-9\\">
      <!-- Only show the two options when web client is authenticated-->
      <!--v-if-->
    </div>
    <div class=\\"col-span-4\\" data-test=\\"feed_filter_component\\">
      <feeddropdownfilter data-test=\\"feed_dropdown_filter\\"></feeddropdownfilter>
    </div>
  </div><!-- IF the following buttons is pressed-->
  <!--v-if-->
</div>"
`;
